pkg/tsdb/cloudwatch/request_parser.go

```diff
@@ -85,6 +86,55 @@ func migrateLegacyQuery(queries []backend.DataQuery, startTime time.Time, endTim
 	return migratedQueries, nil
 }
 
+// migrateStatisticsToStatistic migrates queries that has a `statistics` field to use the `statistic` field instead.
+// In case the query used more than one stat, the first stat in the slice will be used in the statistic field
+// Read more here https://github.com/grafana/grafana/issues/30629
+func migrateStatisticsToStatistic(queryJson *simplejson.Json) (*simplejson.Json, error) {
+	_, err := queryJson.Get("statistic").String()
+	// If there's not a statistic property in the json, we know it's the legacy format and then it has to be migrated
+	if err != nil {
+		stats, err := queryJson.Get("statistics").StringArray()
+		if err != nil {
+			return nil, fmt.Errorf("query must have either statistic or statistics field")
+		}
+		queryJson.Del("statistics")
+		queryJson.Set("statistic", stats[0])
+	}
+
+	return queryJson, nil
+}
+
+var aliasPatterns = map[string]string{
+	"metric":    `${PROP('MetricName')}`,
+	"namespace": `${PROP('Namespace')}`,
+	"period":    `${PROP('Period')}`,
+	"region":    `${PROP('Region')}`,
+	"stat":      `${PROP('Stat')}`,
+	"label":     `${LABEL}`,
+}
+
+var legacyAliasRegexp = regexp.MustCompile(`{{\s*(.+?)\s*}}`)
+
+func migrateAliasToDynamicLabel(queryJson *simplejson.Json) {
+	fullAliasField := queryJson.Get("alias").MustString()
+	if fullAliasField != "" {
+		matches := legacyAliasRegexp.FindAllStringSubmatch(fullAliasField, -1)
+		for _, m := range matches {
+			aliasPattern := m[0]
+			alias := m[1]
```

Comment:
If accessing the items like this 100% safe? Would it make sense to add something like this?
```
if len(m) < 2  { continue } 
```

Reply:
I think so yes since it's based on the return value of the `FindAllStringSubmatch` method

Reply:
>if len(m) < 2  { continue } 

We could add this, but I don't think we could imagine a (test) case that could enter it, and that's why I think that it is might be overly cautious/"paranoid" to add it. 
I could very well be wrong and will discuss that with you.